codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/Encoder.java

```diff
@@ -304,9 +305,7 @@ private HeaderEntry getEntry(CharSequence name, CharSequence value) {
         int h = hash(name);
         int i = index(h);
         for (HeaderEntry e = headerFields[i]; e != null; e = e.next) {
-            if (e.hash == h &&
-                    HpackUtil.equals(name, e.name) &&
-                    HpackUtil.equals(value, e.value)) {
+            if (e.hash == h && (equalsConstantTime(name, e.name) & equalsConstantTime(value, e.value)) != 0) {
```

Comment:
@carl-mastrangelo - note this change ... before it was skipping out early if the name didn't match ... which may yield some timing info.

I propose we do a follow up investigation if you think we can reduce the amount of "constant time comparisons" that we must do.